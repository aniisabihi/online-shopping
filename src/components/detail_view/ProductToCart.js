// noinspection JSUnresolvedVariable

import React from "react";
import { useDispatch } from "react-redux";
import { addToCart } from "../../store/reducers/slices/cartSlice";
import { toast } from "react-toastify";

export default function ProductToCart(props) {
  const available = props.product;
  const color = props.color;
  const power = props.power;
  const storage = props.storage;
  const quantity = props.quantity;
  const dispatch = useDispatch();

  const handleAddToCart = (product) => {
    if (
      product.available &&
      color !== "" &&
      (power !== 0 || storage !== 0) &&
      quantity !== 0
    ) {
      product.color = color;
      product.power = power;
      product.storage = storage;
      product.quantity = quantity;
      dispatch(addToCart(product));
    } else {
      toast.error(
        "Select all product options to be able to add product to cart.",
        { position: "bottom-left" }
      );
    }
  };

  return (
    <button
      disabled={!available}
      id="add-button"
      title="Add to cart"
      aria-label="Add product to cart"
      onClick={() => handleAddToCart(props.product)}
      className="btn-large waves-effect waves-light waves-green"
    >
      <span>Add to cart</span>
    </button>
  );
}
