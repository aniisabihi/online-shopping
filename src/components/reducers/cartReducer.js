import items from './../../products.json';

const initState = {
    items: [items],
    addedItems: [],
    total: 0
}

const cartReducer= (state = initState, action) => {

    return state;
}

export default cartReducer;
