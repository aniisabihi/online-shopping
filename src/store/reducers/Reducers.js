import { combineReducers } from "redux";
import cartReducer from "./slices/cartSlice";

const data = require("./../../data.json");

export const reducers = combineReducers({
  products: data,
  cart: cartReducer,
});

export default data;
