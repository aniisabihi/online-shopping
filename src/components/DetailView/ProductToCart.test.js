import React from "react";
import { shallow } from "enzyme";
import ProductToCart from "./ProductToCart";
import { configureStore } from "@reduxjs/toolkit";
import { reducers } from "../../store/reducers/Reducers";
import { Provider } from "react-redux";

const ProductToCartComponent = <ProductToCart />;
const store = configureStore({ reducer: reducers });

let productToCart = shallow(
  <Provider store={store}>{ProductToCartComponent}</Provider>
);

/* RENDER COMPONENT */
describe("Render ProductToCart", () => {
  it("should render the ProductToCart Component accordingly", () => {
    expect(productToCart).toMatchSnapshot();
  });
});
