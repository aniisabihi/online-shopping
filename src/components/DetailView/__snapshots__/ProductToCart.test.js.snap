// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProductToCart should render the ProductToCart Component accordingly 1`] = `ShallowWrapper {}`;
